rejson.go

```diff
@@ -71,6 +77,20 @@ func (r *Handler) JSONSet(key string, path string, obj interface{}, opts ...rjs.
 	return r.implementation.JSONSet(key, path, obj, opts...)
 }
 
+// JSONSetWithIndex used to set a json object
+//
+// ReJSON syntax:
+// 	JSON.SET <key> <path> <json> <index>
+//
+func (r *Handler) JSONSetWithIndex(key string, path string, obj interface{}, index string) (
+	res interface{}, err error) {
+
+	if r.clientName == rjs.ClientInactive {
+		return nil, rjs.ErrNoClientSet
+	}
+	return r.implementation.JSONSetWithIndex(key, path, obj, index)
+}
+
```

Comment:
Create a separate handler for the ReJSON2, and implement the `ReJSON2` interface on it.
You can also wrap the original handler and create a new handler for ReJSON2.